lib/_http_server.js

```diff
@@ -475,7 +475,12 @@ function socketOnError(e) {
 function onParserExecuteCommon(server, socket, parser, state, ret, d) {
   resetSocketTimeout(server, socket, state);
 
+  if (!d) {
+    d = parser.getCurrentBuffer();
+  }
```

Comment:
I think @bnoordhuis’ comment was basically about undoing my earlier comment (which is probably better, yes). i.e. move this part back into the branches were `d` is actually used, not before the `if`